[Automation] javac compile fails for org.apache.maven.plugins:maven-surefire-plugin:3.0.0

## javac compile on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=3.0.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:2.22.2`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26183264197
Last 300 lines of the log:
```console
Parsed versions: 3.0.0 3.1.0 3.1.2 3.2.1 3.2.2 3.2.3 3.2.5 3.3.0 3.3.1 3.4.0 3.5.0 3.5.1 3.5.2 3.5.3 3.5.4 3.5.5
Timeout per Gradle invocation: 5m
========================================================================================
 Testing org.apache.maven.plugins:maven-surefire-plugin:2.22.2:3.0.0
========================================================================================
========================================================================================
 org.apache.maven.plugins:maven-surefire-plugin:2.22.2:3.0.0 stage native-image run
========================================================================================
Starting stage 'native-image run' for 3.0.0 at 2026-05-20T19:01:54Z
Command: env GVM_TCK_LV=3.0.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:2.22.2
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 6s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.apache.maven.plugins:maven-surefire-plugin:2.22.2
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:14: error: cannot find symbol
import org.apache.maven.plugin.surefire.HelpMojo;
                                       ^
  symbol:   class HelpMojo
  location: package org.apache.maven.plugin.surefire
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:23: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
        ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:23: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
                            ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
3 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:14: error: cannot find symbol
  import org.apache.maven.plugin.surefire.HelpMojo;
                                         ^
    symbol:   class HelpMojo
    location: package org.apache.maven.plugin.surefire
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:23: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
          ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:23: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
                              ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  3 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 5s
13 actionable tasks: 5 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.apache.maven.plugins:maven-surefire-plugin:2.22.2 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 14s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 3.0.0 at 2026-05-20T19:02:08Z with exit code 1 after 14s
❌ ERROR: 'native-image run' for 3.0.0 failed with exit code 1.
SOFT-FAIL[native-image run][3.0.0][env GVM_TCK_LV=3.0.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:2.22.2]
========================================================================================
 org.apache.maven.plugins:maven-surefire-plugin:2.22.2:3.0.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 3.0.0 at 2026-05-20T19:02:08Z
Command: env GVM_TCK_LV=3.0.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:2.22.2
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 886ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:14: error: cannot find symbol
import org.apache.maven.plugin.surefire.HelpMojo;
                                       ^
  symbol:   class HelpMojo
  location: package org.apache.maven.plugin.surefire
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:23: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
        ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:23: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
                            ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
3 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:14: error: cannot find symbol
  import org.apache.maven.plugin.surefire.HelpMojo;
                                         ^
    symbol:   class HelpMojo
    location: package org.apache.maven.plugin.surefire
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:23: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
          ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:23: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
                              ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  3 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 3.0.0 at 2026-05-20T19:02:11Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 3.0.0 failed with exit code 1.
FAILED[javac compile][3.0.0][env GVM_TCK_LV=3.0.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:2.22.2]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=3.0.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:2.22.2`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26183264197
Last 300 lines of the log:
```console
Parsed versions: 3.0.0 3.1.0 3.1.2 3.2.1 3.2.2 3.2.3 3.2.5 3.3.0 3.3.1 3.4.0 3.5.0 3.5.1 3.5.2 3.5.3 3.5.4 3.5.5
Timeout per Gradle invocation: 5m
========================================================================================
 Testing org.apache.maven.plugins:maven-surefire-plugin:2.22.2:3.0.0
========================================================================================
========================================================================================
 org.apache.maven.plugins:maven-surefire-plugin:2.22.2:3.0.0 stage native-image run
========================================================================================
Starting stage 'native-image run' for 3.0.0 at 2026-05-20T19:01:44Z
Command: env GVM_TCK_LV=3.0.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:2.22.2
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 6s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.apache.maven.plugins:maven-surefire-plugin:2.22.2
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:14: error: cannot find symbol
import org.apache.maven.plugin.surefire.HelpMojo;
                                       ^
  symbol:   class HelpMojo
  location: package org.apache.maven.plugin.surefire
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:23: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
        ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:23: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
                            ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
3 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:14: error: cannot find symbol
  import org.apache.maven.plugin.surefire.HelpMojo;
                                         ^
    symbol:   class HelpMojo
    location: package org.apache.maven.plugin.surefire

  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:23: error: cannot find symbol
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

          HelpMojo mojo = new HelpMojo();
          ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:23: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
                              ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  3 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 5s
For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 5 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.apache.maven.plugins:maven-surefire-plugin:2.22.2 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 13s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 3.0.0 at 2026-05-20T19:01:58Z with exit code 1 after 14s
❌ ERROR: 'native-image run' for 3.0.0 failed with exit code 1.
SOFT-FAIL[native-image run][3.0.0][env GVM_TCK_LV=3.0.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:2.22.2]
========================================================================================
 org.apache.maven.plugins:maven-surefire-plugin:2.22.2:3.0.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 3.0.0 at 2026-05-20T19:01:58Z
Command: env GVM_TCK_LV=3.0.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:2.22.2
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 851ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:14: error: cannot find symbol
import org.apache.maven.plugin.surefire.HelpMojo;
                                       ^
  symbol:   class HelpMojo
  location: package org.apache.maven.plugin.surefire
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:23: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
        ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:23: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
                            ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
3 errors

FAILURE: Build failed with an exception.

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/build/reports/problems/problems-report.html

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:14: error: cannot find symbol
  import org.apache.maven.plugin.surefire.HelpMojo;
                                         ^
    symbol:   class HelpMojo
    location: package org.apache.maven.plugin.surefire
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:23: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
          ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:23: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
                              ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  3 errors


Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date
* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 3.0.0 at 2026-05-20T19:02:01Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 3.0.0 failed with exit code 1.
FAILED[javac compile][3.0.0][env GVM_TCK_LV=3.0.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:2.22.2]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `env GVM_TCK_LV=3.0.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:2.22.2`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26183264197
Last 300 lines of the log:
```console
Parsed versions: 3.0.0 3.1.0 3.1.2 3.2.1 3.2.2 3.2.3 3.2.5 3.3.0 3.3.1 3.4.0 3.5.0 3.5.1 3.5.2 3.5.3 3.5.4 3.5.5
Timeout per Gradle invocation: 5m
========================================================================================
 Testing org.apache.maven.plugins:maven-surefire-plugin:2.22.2:3.0.0
========================================================================================
========================================================================================
 org.apache.maven.plugins:maven-surefire-plugin:2.22.2:3.0.0 stage native-image run
========================================================================================
Starting stage 'native-image run' for 3.0.0 at 2026-05-20T19:01:44Z
Command: env GVM_TCK_LV=3.0.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:2.22.2
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.apache.maven.plugins:maven-surefire-plugin:2.22.2
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:14: error: cannot find symbol
import org.apache.maven.plugin.surefire.HelpMojo;
                                       ^
  symbol:   class HelpMojo
  location: package org.apache.maven.plugin.surefire
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:23: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
        ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:23: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
                            ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
3 errors

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 5 executed, 8 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:14: error: cannot find symbol
  import org.apache.maven.plugin.surefire.HelpMojo;
                                         ^
    symbol:   class HelpMojo
    location: package org.apache.maven.plugin.surefire
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:23: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
          ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:23: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
                              ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  3 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 5s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.apache.maven.plugins:maven-surefire-plugin:2.22.2 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 14s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 3.0.0 at 2026-05-20T19:01:58Z with exit code 1 after 14s
❌ ERROR: 'native-image run' for 3.0.0 failed with exit code 1.
SOFT-FAIL[native-image run][3.0.0][env GVM_TCK_LV=3.0.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:2.22.2]
========================================================================================
 org.apache.maven.plugins:maven-surefire-plugin:2.22.2:3.0.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 3.0.0 at 2026-05-20T19:01:58Z
Command: env GVM_TCK_LV=3.0.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:2.22.2
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 856ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:14: error: cannot find symbol
import org.apache.maven.plugin.surefire.HelpMojo;
                                       ^
  symbol:   class HelpMojo
  location: package org.apache.maven.plugin.surefire
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:23: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
        ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:23: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
                            ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
3 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.

> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:14: error: cannot find symbol
  import org.apache.maven.plugin.surefire.HelpMojo;
                                         ^
    symbol:   class HelpMojo
    location: package org.apache.maven.plugin.surefire
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:23: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
          ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/2.22.2/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:23: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
                              ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  3 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 3.0.0 at 2026-05-20T19:02:01Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 3.0.0 failed with exit code 1.
FAILED[javac compile][3.0.0][env GVM_TCK_LV=3.0.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:2.22.2]
```

--- a/metadata/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/reachability-metadata.json
+++ b/metadata/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/reachability-metadata.json
@@ -1,0 +1,16 @@
+{
+  "resources": [
+    {
+      "condition": {
+        "typeReached": "org.apache.maven.plugins.maven_surefire_plugin.HelpMojo"
+      },
+      "glob": "META-INF/maven/org.apache.maven.plugins/maven-surefire-plugin/plugin-help.xml"
+    },
+    {
+      "condition": {
+        "typeReached": "org.apache.maven.plugins.maven_surefire_plugin.HelpMojo"
+      },
+      "glob": "META-INF/services/javax.xml.parsers.DocumentBuilderFactory"
+    }
+  ]
+}

--- a/metadata/org.apache.maven.plugins/maven-surefire-plugin/index.json
+++ b/metadata/org.apache.maven.plugins/maven-surefire-plugin/index.json
@@ -11,7 +11,7 @@
       "3.0.0"
     ],
     "allowed-packages" : [
-      "org.apache.maven.plugin.surefire"
+      "org.apache.maven.plugins.maven_surefire_plugin"
     ]
   },
   {

--- a/metadata/org.apache.maven.plugins/maven-surefire-plugin/index.json
+++ b/metadata/org.apache.maven.plugins/maven-surefire-plugin/index.json
@@ -2,6 +2,11 @@
   {
     "latest" : true,
     "metadata-version" : "3.0.0",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-surefire-plugin/$version$/maven-surefire-plugin-$version$-sources.jar",
+    "repository-url" : "https://github.com/apache/maven-surefire",
+    "test-code-url" : "https://repo.maven.apache.org/maven2/org/apache/maven/surefire/surefire/$version$/surefire-$version$-source-release.zip",
+    "documentation-url" : "https://maven.apache.org/surefire-archives/surefire-$version$/maven-surefire-plugin/",
+    "description" : "The Maven Surefire Plugin runs a Maven project's unit tests during the test phase and reports the results. It integrates test providers such as JUnit, TestNG, and POJO-based tests with Maven's lifecycle, including options for filtering, forking, and failure handling.",
     "tested-versions" : [
       "3.0.0"
     ],

--- a/metadata/org.apache.maven.plugins/maven-surefire-plugin/index.json
+++ b/metadata/org.apache.maven.plugins/maven-surefire-plugin/index.json
@@ -1,6 +1,15 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "3.0.0",
+    "tested-versions" : [
+      "3.0.0"
+    ],
+    "allowed-packages" : [
+      "org.apache.maven.plugin.surefire"
+    ]
+  },
+  {
     "metadata-version" : "2.22.2",
     "source-code-url" : "https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-surefire-plugin/$version$/maven-surefire-plugin-$version$-sources.jar",
     "repository-url" : "https://github.com/apache/maven-surefire",

--- a/stats/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/execution-metrics.json
+++ b/stats/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/execution-metrics.json
@@ -1,0 +1,99 @@
+{
+  "fix_javac_fail:2026-06-07": {
+    "timestamp": "2026-06-07T16:20:29.710671Z",
+    "library": "org.apache.maven.plugins:maven-surefire-plugin:3.0.0",
+    "starting_commit": "e412b920b1fc38c4848220ffa5d9f5a693031e40",
+    "ending_commit": "9e28e104d70dfa30d32188f3d3a4444ac4c13cec",
+    "previous_library": "org.apache.maven.plugins:maven-surefire-plugin:2.22.2",
+    "strategy_name": "javac_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "3.0.0",
+      "dynamicAccess": {
+        "breakdown": {
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 1,
+            "totalCalls": 1
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 1,
+        "totalCalls": 1
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 538,
+          "missed": 670,
+          "ratio": 0.445364,
+          "total": 1208
+        },
+        "line": {
+          "covered": 103,
+          "missed": 171,
+          "ratio": 0.375912,
+          "total": 274
+        },
+        "method": {
+          "covered": 14,
+          "missed": 83,
+          "ratio": 0.14433,
+          "total": 97
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "2.22.2",
+      "dynamicAccess": {
+        "breakdown": {
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 1,
+            "totalCalls": 1
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 1,
+        "totalCalls": 1
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 559,
+          "missed": 596,
+          "ratio": 0.483983,
+          "total": 1155
+        },
+        "line": {
+          "covered": 106,
+          "missed": 141,
+          "ratio": 0.42915,
+          "total": 247
+        },
+        "method": {
+          "covered": 14,
+          "missed": 64,
+          "ratio": 0.179487,
+          "total": 78
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 16089,
+      "cached_input_tokens_used": 100864,
+      "output_tokens_used": 971,
+      "iterations": 1,
+      "cost_usd": 0.0795,
+      "tested_library_loc": 103,
+      "metadata_entries": 2,
+      "previous_library_metadata_entries": 2,
+      "code_coverage_percent": 37.59,
+      "previous_library_coverage_percent": 42.91
+    },
+    "artifacts": {
+      "test_file": "tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java",
+      "metadata_file": "metadata/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/stats.json
+++ b/stats/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/stats.json
@@ -1,0 +1,37 @@
+{
+  "versions" : [ {
+    "version" : "3.0.0",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "resources" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 1,
+          "totalCalls" : 1
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 1,
+      "totalCalls" : 1
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 538,
+        "missed" : 670,
+        "ratio" : 0.445364,
+        "total" : 1208
+      },
+      "line" : {
+        "covered" : 103,
+        "missed" : 171,
+        "ratio" : 0.375912,
+        "total" : 274
+      },
+      "method" : {
+        "covered" : 14,
+        "missed" : 83,
+        "ratio" : 0.14433,
+        "total" : 97
+      }
+    }
+  } ]
+}

--- a/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/.gitignore
+++ b/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/build.gradle
+++ b/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/build.gradle
@@ -1,0 +1,27 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "org.apache.maven.plugins:maven-surefire-plugin:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/gradle.properties
+++ b/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = org.apache.maven.plugins:maven-surefire-plugin:3.0.0
+library.version = 3.0.0
+metadata.dir = org.apache.maven.plugins/maven-surefire-plugin/3.0.0/

--- a/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/settings.gradle
+++ b/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'org.apache.maven.plugins.maven-surefire-plugin_tests'

--- a/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java
+++ b/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java
@@ -1,0 +1,54 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_apache_maven_plugins.maven_surefire_plugin;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.logging.SystemStreamLog;
+import org.apache.maven.plugin.surefire.HelpMojo;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class HelpMojoTest {
+
+    @Test
+    void executeLoadsPluginHelpResource() throws MojoExecutionException {
+        HelpMojo mojo = new HelpMojo();
+        RecordingLog log = new RecordingLog();
+        mojo.setLog(log);
+
+        mojo.execute();
+
+        assertThat(log.infoMessages)
+                .anySatisfy(message -> assertThat(message)
+                        .contains("Surefire")
+                        .contains("surefire:test"));
+    }
+
+    private static final class RecordingLog extends SystemStreamLog {
+
+        private final List<String> infoMessages = new ArrayList<>();
+
+        @Override
+        public boolean isInfoEnabled() {
+            return true;
+        }
+
+        @Override
+        public void info(CharSequence content) {
+            infoMessages.add(content.toString());
+        }
+
+        @Override
+        public void info(CharSequence content, Throwable error) {
+            info(content);
+        }
+    }
+}

--- a/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java
+++ b/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java
@@ -11,7 +11,7 @@ import java.util.List;
 
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugin.logging.SystemStreamLog;
-import org.apache.maven.plugin.surefire.HelpMojo;
+import org.apache.maven.plugins.maven_surefire_plugin.HelpMojo;
 import org.junit.jupiter.api.Test;
 
 import static org.assertj.core.api.Assertions.assertThat;

--- a/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/user-code-filter.json
+++ b/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/user-code-filter.json
@@ -1,0 +1,13 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "org.apache.maven.plugin.surefire.**"
+    },
+    {
+      "includeClasses" : "org_apache_maven_plugins.maven_surefire_plugin.**"
+    }
+  ]
+}

--- a/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/user-code-filter.json
+++ b/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/user-code-filter.json
@@ -7,6 +7,9 @@
       "includeClasses" : "org.apache.maven.plugin.surefire.**"
     },
     {
+      "includeClasses" : "org.apache.maven.plugins.maven_surefire_plugin.**"
+    },
+    {
       "includeClasses" : "org_apache_maven_plugins.maven_surefire_plugin.**"
     }
   ]

